clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -17151,28 +17168,6 @@ void ComplexAPIRule::runRule(
 
 REGISTER_RULE(ComplexAPIRule)
 
-void TemplateSpecializationTypeLocRule::registerMatcher(
```

Comment:
Is it better to keep `TemplateSpecializationTypeLocRule` here? Maybe other types can reuse it (I have a WIP PR used this rule).

Reply:
It's reasonable to keep it,  I will remove the migration implementation of atomic template class and keep this rule. I'm worried that too many implementations written in it will lead to too long and cumbersome files. 
And when I modified this rule, there was no other migration implementation, so I removed it. 

This rule will be keeped as an empty rule in my PR.